app/js/elements/fxos-inspector.js

```diff
@@ -0,0 +1,485 @@
+      return;
+    }
+
+    this.render();
+  });
+};
+
+proto.render = function() {
+  clearTimeout(this._searchTimeout);
+
+  this.target = this.rootTarget;
+  this.keyPath.forEach(key => this.target = this.target[key]);
+
+  this.container.innerHTML = renderTargetPage(this.target, this.keyPath.length === 0).innerHTML;
+  this.container.scrollTo(0, 0);
+};
+
+proto.setRootTarget = function(rootTarget) {
+  this.rootTarget = rootTarget;
+  this.target = this.rootTarget;
+
+  this.keyPath = [];
+  this.render();
+};
+
+proto._handleButtonClick = function(target) {
+  switch (target.dataset.action) {
+    case 'cancel':
+      this.keyPath.pop();
+      break;
+    case 'save':
+      (() => {
+        var value = this.container.querySelector('fxos-code-editor').value;
+        var target = this.rootTarget;
+
+        this.keyPath.forEach((key, index) => {
+          if (index < this.keyPath.length - 1) {
+            target = target[key];
+            return;
+          }
+
+          target[key] = value;
+        });
+
+        this.dispatchEvent(new CustomEvent('save', {
+          detail: JSON.stringify({
+            keyPath: this.keyPath,
+            expression: '[\'' + this.keyPath.join('\'][\'') + '\']',
+            value: value
+          })
+        }));
+
+        this.keyPath.pop();
+      })();
+      break;
+    default:
+      break;
+  }
+
+  this.render();
+};
+
+proto._handleRemoveAttributeClick = function(target) {
+  var name = target.getAttribute('href');
+  this.target.removeNamedItem(name);
+
+  this.dispatchEvent(new CustomEvent('removeattribute', {
+    detail: JSON.stringify({
+      keyPath: this.keyPath,
+      expression: '[\'' + this.keyPath.join('\'][\'') + '\']',
+      name: name
+    })
+  }));
+
+  this.render();
+};
+
+proto._handleListItemClick = function(target) {
+  // If the link clicked has an `href` pointing
+  // to "../", go back to the previous level in
+  // the key path.
+  var href = target.getAttribute('href');
+  if (href === '../') {
+    this.keyPath.pop();
+  }
+
+  // Show "Create New Attribute" prompt.
+  else if (href === '#add-attribute') {
+    this.newAttributeName.open();
+  }
+
+  // Otherwise, push the next part of the key path.
+  else {
+    this.keyPath.push(href);
+  }
+
+  this.render();
+};
+
+/**
+ * Helper function for determining the type of
+ * page to render for the current target (editor
+ * or list).
+ */
+function renderTargetPage(target, isRoot) {
+  // If the target is not an object, assume that
+  // it is editable and render the editor page.
+  if (typeof target !== 'object') {
+    return renderTargetEditor(target);
+  }
+
+  // Otherwise, the target is an enumerable object
+  // and we can render the list page.
+  return renderTargetList(target, isRoot);
+}
+
+/**
+ * Helper for rendering the editor page.
+ */
+function renderTargetEditor(target) {
+  var page = document.createElement('section');
+  page.innerHTML =
+`<fxos-code-editor>${target}</fxos-code-editor>
```

Comment:
This actually isn't slow. The <fxos-code-editor> component is very lightweight. Not much more than just a <textarea> with some additional decoration/styling.